aiida/tools/dbimporters/plugins/icsd.py

```diff
@@ -579,6 +589,9 @@ def query_page(self):
                 self.cursor.execute('SELECT FOUND_ROWS()')
                 self.number_of_results = int(self.cursor.fetchone()[0])
 
+            for idnum in self._results:
+                self.is_theoretical.append(self.query_db_is_theoretical(idnum))
```

Comment:
I wonder about the efficiency of this approach. If I understand correctly, you are executing another dedicated query for each matched CIF to retrieve the remarks. I think it should be possible to directly include this in the main query on line 578, wouldn't it? This would greatly reduce the number of queries and efficiency.

Reply:
Should be able to `JOIN` the `SELECT` on `icsd_remarks` and get a boolean for whether it contains `THE` or `ZTHE` straight in SQL.

Reply:
Good idea, thanks!